src/components/password-modal/password-modal-component.tsx

```diff
@@ -152,19 +152,32 @@ export class PasswordModalComponent extends React.Component<
         return this.resultDeferred.promise;
     }
 
-    public createPassword(subtitle?: string) {
+    public async createPassword(subtitle?: string) {
         this.resultDeferred = new Deferred();
         this.clearErrorMessage();
 
-        this.setState({
-            visible: true,
-            title: translate('Password.setupPinTitle'),
-            subtitle: subtitle || translate('Password.createPinSubtitle'),
-            currentStep: ScreenStep.CREATE_PIN_TERMS,
-            enableBiometryAuth: false,
-            allowBackButton: true,
-            showAttempts: false
-        });
+        const passwordCredentials = await getPassword();
+        if (passwordCredentials && passwordCredentials.password) {
+            this.setState({
```

Comment:
this doesn't seem to be right :)
create password should create password, we shouldn't start this flow if a password is already present, or much better this flow should return a rejection is there is a password already set